x-pack/plugins/lens/public/indexpattern_datasource/loader.ts

```diff
@@ -82,20 +88,67 @@ const setLastUsedIndexPatternId = (storage: IStorageWrapper, value: string) => {
   writeToStorage(storage, 'indexPatternId', value);
 };
 
+const CURRENT_PATTERN_REFERENCE_NAME = 'indexpattern-datasource-current-indexpattern';
```

Comment:
I'm not sure I see the reasoning behind this- why do we add a reference to the selected index pattern, isn't this only part of the local state, not the persisted state?